In [4]:
# Day 6 — Data Loading
# Importing the libraries we need today

import pandas as pd
import geopandas as gpd

In [2]:
# Loading the NFHS-5 district dataset
df = pd.read_csv("../data/raw/nfhs5_district_kaggle.csv")

print("Shape:", df.shape)

Shape: (706, 109)


In [3]:
print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

Rows: 706
Columns: 109


In [4]:
# See the first 5 rows of the dataset
df.head()

,District Names,State/UT,Number of Households surveyed,Number of Women age 15-49 years interviewed,Number of Men age 15-54 years interviewed,Female population age 6 years and above who ever attended school (%),Population below age 15 years (%),"Sex ratio of the total population (females per 1,000 males)","Sex ratio at birth for children born in the last five years (females per 1,000 males)",Children under age 5 years whose birth was registered with the civil authority (%),...,Men age 15 years and above wih Mildly elevated blood pressure (Systolic 140-159 mm of Hg and/or Diastolic 90-99 mm of Hg) (%),Men age 15 years and above wih Moderately or severely elevated blood pressure (Systolic ≥160 mm of Hg and/or Diastolic ≥100 mm of Hg) (%),Men age 15 years and above wih Elevated blood pressure (Systolic ≥140 mm of Hg and/or Diastolic ≥90 mm of Hg) or taking medicine to control blood pressure (%),Women (age 30-49 years) Ever undergone a screening test for cervical cancer (%),Women (age 30-49 years) Ever undergone a breast examination for breast cancer (%),Women (age 30-49 years) Ever undergone an oral cavity examination for oral cancer (%),Women age 15 years and above who use any kind of tobacco (%),Men age 15 years and above who use any kind of tobacco (%),Women age 15 years and above who consume alcohol (%),Men age 15 years and above who consume alcohol (%)
0,Nicobars,Andaman & Nicobar Islands,882,764,125,78.0,23.0,973,927,98.0,...,32.9,11.1,47.0,13.4,13.2,5.4,63.5,76.8,29.6,64.5
1,North & Middle Andaman,Andaman & Nicobar Islands,874,789,108,82.7,19.8,950,844,100.0,...,22.6,6.0,32.2,1.7,0.3,15.8,46.8,70.5,5.1,45.3
2,South Andaman,Andaman & Nicobar Islands,868,844,134,84.7,21.0,967,935,96.5,...,17.9,6.1,26.9,1.3,0.7,8.0,19.6,50.8,1.7,32.8
3,Srikakulam,Andhra Pradesh,874,780,100,60.0,20.7,1140,1163,95.0,...,14.4,5.5,22.9,1.0,0.2,3.8,7.1,21.3,0.6,28.3
4,Vizianagaram,Andhra Pradesh,902,853,134,56.0,20.6,1114,898,95.4,...,14.8,6.4,25.1,4.9,0.6,7.3,11.4,21.5,0.8,32.3


In [5]:
# See all column names
df.columns.tolist()

['District Names',
 'State/UT',
 'Number of Households surveyed',
 'Number of Women age 15-49 years interviewed',
 'Number of Men age 15-54 years interviewed',
 'Female population age 6 years and above who ever attended school (%)',
 'Population below age 15 years (%)',
 ' Sex ratio of the total population (females per 1,000 males)',
 'Sex ratio at birth for children born in the last five years (females per 1,000 males)',
 'Children under age 5 years whose birth was registered with the civil authority (%)',
 'Deaths in the last 3 years registered with the civil authority (%)',
 'Population living in households with electricity (%)',
 'Population living in households with an improved drinking-water source1 (%)',
 'Population living in households that use an improved sanitation facility2 (%)',
 'Households using clean fuel for cooking3 (%)',
 'Households using iodized salt (%)',
 'Households with any usual member covered under a health insurance/financing scheme (%)',
 'Children age 5 ye

In [6]:
# Check data types of all columns
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 706 entries, 0 to 705
Columns: 109 entries, District Names to Men age 15 years and above who consume alcohol (%)
dtypes: float64(54), int64(4), object(51)
memory usage: 601.3+ KB


In [7]:
# Check missing values in every column
df.isnull().sum()

District Names                                                                           0
State/UT                                                                                 0
Number of Households surveyed                                                            0
Number of Women age 15-49 years interviewed                                              0
Number of Men age 15-54 years interviewed                                                0
                                                                                        ..
Women (age 30-49 years) Ever undergone an oral cavity examination for oral cancer (%)    0
Women age 15 years and above who use any kind of tobacco (%)                             0
Men age 15 years and above who use any kind of tobacco (%)                               0
Women age 15 years and above who consume alcohol (%)                                     0
Men age 15 years and above who consume alcohol (%)                                       0

In [8]:
# Quick statistical summary of numerical columns
df.describe()

,Number of Households surveyed,Number of Women age 15-49 years interviewed,Number of Men age 15-54 years interviewed,Female population age 6 years and above who ever attended school (%),Population below age 15 years (%),"Sex ratio of the total population (females per 1,000 males)",Children under age 5 years whose birth was registered with the civil authority (%),Population living in households with electricity (%),Population living in households with an improved drinking-water source1 (%),Population living in households that use an improved sanitation facility2 (%),...,Men age 15 years and above wih Mildly elevated blood pressure (Systolic 140-159 mm of Hg and/or Diastolic 90-99 mm of Hg) (%),Men age 15 years and above wih Moderately or severely elevated blood pressure (Systolic ≥160 mm of Hg and/or Diastolic ≥100 mm of Hg) (%),Men age 15 years and above wih Elevated blood pressure (Systolic ≥140 mm of Hg and/or Diastolic ≥90 mm of Hg) or taking medicine to control blood pressure (%),Women (age 30-49 years) Ever undergone a screening test for cervical cancer (%),Women (age 30-49 years) Ever undergone a breast examination for breast cancer (%),Women (age 30-49 years) Ever undergone an oral cavity examination for oral cancer (%),Women age 15 years and above who use any kind of tobacco (%),Men age 15 years and above who use any kind of tobacco (%),Women age 15 years and above who consume alcohol (%),Men age 15 years and above who consume alcohol (%)
count,706.000000,706.000000,706.000000,706.000000,706.000000,706.000000,706.000000,706.000000,706.000000,706.000000,...,706.000000,706.000000,706.000000,706.000000,706.000000,706.000000,706.000000,706.000000,706.000000,706.000000
mean,900.481586,1023.929178,143.975921,71.514448,26.367847,1020.640227,91.056516,97.001700,93.722946,71.932153,...,16.264873,6.052550,24.774788,1.574363,0.657649,0.705949,11.622238,40.591926,2.922380,23.195184
std,69.320330,177.174124,31.967905,10.319860,5.297820,73.400047,9.392669,4.354679,8.718210,14.290983,...,4.337494,2.572881,6.767158,2.774812,1.569643,1.468871,11.950115,14.087173,6.083796,13.373136
min,213.000000,216.000000,17.000000,45.400000,16.000000,755.000000,51.600000,68.400000,41.200000,29.200000,...,5.300000,0.800000,10.000000,0.000000,0.000000,0.000000,0.100000,6.800000,0.000000,0.100000
25%,882.000000,911.000000,124.000000,64.400000,22.500000,969.000000,87.025000,96.400000,92.000000,62.000000,...,13.200000,4.100000,19.800000,0.200000,0.000000,0.000000,4.100000,30.350000,0.300000,13.600000
50%,908.000000,1020.000000,145.000000,71.350000,25.400000,1013.000000,94.850000,98.700000,97.000000,73.750000,...,16.300000,5.800000,24.400000,0.600000,0.200000,0.300000,7.700000,42.500000,0.500000,20.150000
75%,931.000000,1141.500000,164.000000,78.975000,29.500000,1065.750000,97.700000,99.500000,99.300000,83.300000,...,18.875000,7.600000,29.000000,1.500000,0.500000,0.700000,14.800000,50.975000,1.700000,30.900000
max,990.000000,1621.000000,241.000000,99.200000,50.600000,1332.000000,100.000000,100.000000,100.000000,99.900000,...,32.900000,19.500000,49.600000,23.200000,14.600000,15.800000,70.600000,80.600000,42.800000,68.400000


In [9]:
# Check our 3 target columns specifically
print(df['Children under 5 years who are stunted (height-for-age)18 (%)'].dtype)
print(df['Children under 5 years who are wasted (weight-for-height)18 (%)'].dtype)
print(df['Children under 5 years who are underweight (weight-for-age)18 (%)'].dtype)

object
object
object


In [5]:
# Load the India district shapefile
gdf = gpd.read_file("../data/raw/india_shapefile/gadm41_IND_2.shp")

print("Shape:", gdf.shape)
print("Columns:", gdf.columns.tolist())

Shape: (676, 14)
Columns: ['GID_2', 'GID_0', 'COUNTRY', 'GID_1', 'NAME_1', 'NL_NAME_1', 'NAME_2', 'VARNAME_2', 'NL_NAME_2', 'TYPE_2', 'ENGTYPE_2', 'CC_2', 'HASC_2', 'geometry']


In [6]:
gdf.head()

,GID_2,GID_0,COUNTRY,GID_1,NAME_1,NL_NAME_1,NAME_2,VARNAME_2,NL_NAME_2,TYPE_2,ENGTYPE_2,CC_2,HASC_2,geometry
0,IND.1.1_1,IND,India,IND.1_1,Andaman and Nicobar,NA,Nicobar Islands,NA,NA,District,District,NA,IN.AN.NI,"MULTIPOLYGON (((93.78988 6.85201, 93.79015 6.8..."
1,IND.1.2_1,IND,India,IND.1_1,Andaman and Nicobar,NA,North and Middle Andaman,NA,NA,District,District,NA,IN.AN.NM,"MULTIPOLYGON (((92.84441 12.14969, 92.84466 12..."
2,IND.1.3_1,IND,India,IND.1_1,Andaman and Nicobar,NA,South Andaman,NA,NA,District,District,NA,IN.AN.SA,"MULTIPOLYGON (((92.52111 10.89694, 92.52306 10..."
3,IND.2.1_1,IND,India,IND.2_1,Andhra Pradesh,NA,Anantapur,"Anantpur, Ananthapur",NA,District,District,NA,IN.AD.AN,"POLYGON ((77.846 13.92832, 77.83012 13.92715, ..."
4,IND.2.2_1,IND,India,IND.2_1,Andhra Pradesh,NA,Chittoor,Chitoor|Chittor,NA,District,District,NA,IN.AD.CH,"POLYGON ((78.54555 12.74391, 78.55031 12.73876..."


## Day 6 — Data Loading Observations

### NFHS-5 Dataset
- 706 districts, 109 columns loaded successfully
- 54 float64, 4 int64, 51 object columns
- All 3 target variables found (stunting, wasting, underweight)
- Target columns are object type — need conversion to float in cleaning phase
- isnull().sum() shows 0 missing — but object columns likely hide real missing values as * or -

### Shapefile Dataset
- 676 rows, 14 columns loaded successfully
- Key columns: NAME_2 (district name), NAME_1 (state name), geometry (map boundaries)
- 676 vs 706 districts — mismatch to be resolved during merge in Day 12

### Next: Day 7 — Deep data cleaning begins